apps/backend/src/app/modules/maps/maps.service.ts

```diff
@@ -552,6 +552,15 @@ export class MapsService {
     const hasVmf = vmfFiles?.length > 0;
     const bspHash = FileStoreService.getHashForBuffer(bspFile.buffer);
 
+    // Check for duplicate map hash
+    const existingMap = await this.db.mMap.findUnique({
+      where: { bspHash }
+    });
```

Comment:
Bit of a nitpick but we have the `.exists` client exists for this. Also prefer not using `findUnique` when we don't actually have a DB-level uniqueness constraint on `bspHash`.

Reply:
This is outdated and I started using `findFirst`. Happy to use `.exists` though!